web/web.go

```diff
@@ -65,8 +65,17 @@ type Handler struct {
 	reloadCh    chan struct{}
 	options     *Options
 	statusInfo  *PrometheusStatus
+}
+
+// ApplyConfig updates the status state as the new config requires.
+// Returns true on success.
+func (h *Handler) ApplyConfig(conf *config.Config) bool {
+	h.federation.mtx.Lock()
```

Comment:
Yea, I felt this was just going too deep in terms of abstraction for a single assignment.